tipg/backends/aurora/database.py

```diff
@@ -0,0 +1,85 @@
+"""tipg.db: database events."""
+
+from typing import List, Optional
+
+import orjson
+from buildpg import asyncpg
+
+from tipg.logger import logger
+from tipg.settings import PostgresSettings
+
+from fastapi import FastAPI
+
+
+class connection_factory:
+    """Connection creation."""
+
+    tipg_schema: str
+    schemas: List[str]
+
+    def __init__(self, tipg_schema: str, schemas: Optional[List[str]] = None) -> None:
+        """Init."""
+        self.tipg_schema = tipg_schema
+        self.schemas = schemas or []
+
+    async def __call__(self, conn: asyncpg.Connection):
+        """Create connection."""
+        await conn.set_type_codec(
+            "json", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
+        )
+        await conn.set_type_codec(
+            "jsonb", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
+        )
+
+        schemas = ",".join([self.tipg_schema, *self.schemas])
+
+        logger.debug(f"Looking for Tables and Functions in {schemas} schemas")
+
+        await conn.execute(
+            f"""
+            SELECT set_config(
+                'search_path',
+                '{schemas},' || current_setting('search_path', false),
+                false
+                );
+            """
+        )
```

Comment:
almost like tipg.database.connection_factory but without SQL registering to the pg_temp schema